homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,130 @@
+                hass.config.units.temperature_unit, 'mdi:thermometer',
+                DEVICE_CLASS_TEMPERATURE))
+        if device.displayedOutdoorHumidity:
+            devices.append(LyricSensor(
+                device, location, hass, 'displayedOutdoorHumidity',
+                'Humidity Outside', '%', 'mdi:water-percent',
+                DEVICE_CLASS_HUMIDITY))
+        if device.nextPeriodTime:
+            devices.append(LyricSensor(
+                device, location, hass, 'nextPeriodTime',
+                'Next Period Time', '', 'mdi:clock',
+                DEVICE_CLASS_TIMESTAMP))
+        if device.thermostatSetpointStatus:
+            devices.append(LyricSensor(
+                device, location, hass, 'thermostatSetpointStatus',
+                'Status', '', 'mdi:thermostat', None))
+
+    async_add_entities(devices, True)
+
+
+class LyricSensor(LyricDeviceEntity):
+    """Representation of a Lyric thermostat."""
+
+    def __init__(self, device, location, hass, key, key_name,
+                 unit_of_measurement=None, icon=None,
+                 device_class=None) -> None:
+        """Initialize the sensor."""
+        unique_id = '{}_{}'.format(
+            device.macID, key_name.replace(" ", "_"))
+        self._unit_of_measurement = unit_of_measurement
+        self._state = None
+        self._available = False
+        self.key = key
+
+        name = '{} {}'.format(device.name, key_name)
+
+        super().__init__(device, location, unique_id, name, icon, device_class)
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    async def _lyric_update(self) -> None:
+        """Get values from lyric."""
+        if self.device:
+            if self.key == 'thermostatSetpointStatus':
+                status = getattr(self.device, self.key)
+                if status == 'NoHold':
+                    self._state = 'Following Schedule'
+                elif status == 'HoldUntil':
+                    self._state = 'Held until {}'.format(
+                        self.device.nextPeriodTime[:-3])
+                elif status == 'PermanentHold':
+                    self._state = 'Held Permanently'
+                elif status == 'VacationHold':
+                    self._state = 'Holiday'
+            else:
+                state = getattr(self.device, self.key)
+                if self._device_class == DEVICE_CLASS_TIMESTAMP:
+                    date = datetime.now()
+                    time = datetime.strptime(state, '%H:%M:%S').replace(
+                        day=date.day, month=date.month, year=date.year)
+                    if time <= datetime.now():
+                        date = date + timedelta(days=1)
+                        time = time.replace(day=date.day)
+                    state = time
```

Comment:
When using the timestamp device class, the time should be absolute utc iso formatted.

Reply:
See updated code. This now returns:

![image](https://user-images.githubusercontent.com/28114703/61579581-e8766880-aafe-11e9-8944-a26bd43c20e3.png)

*(ui bug found btw)*